notebooks/02_quant_activation_overview.ipynb

```diff
@@ -664,25 +664,233 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 20,
+   "execution_count": 31,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "out1_train = quant_hard_tanh(inp1)\n",
+    "quant_hard_tanh.eval()\n",
+    "out2_eval = quant_hard_tanh(inp2)\n",
+    "assert out1_train.scale.isclose(out2_eval.scale).item()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In all of the examples that have currently been looked at in this tutorial, we have used per-tensor quantization. I.e., the output tensor of the activation, if quantized, was always quantized on a per-tensor level, with a single scale and zero-point quantization parameter per output tensor. However, one can also do per-channel quantization, where each output channel of the tensor has its own quantization parameters. In the example below, we look at per-tensor quantization of an input tensor that has 3 channels and 256 elements in the height and width dimensions. We purposely mutate the 1st channel to have its dynamic range be 3 times larger than the other 2 channels. We then feed it through a `QuantReLU`, whose default behavior is to quantize at a per-tensor granularity."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 32,
    "metadata": {},
    "outputs": [
     {
      "data": {
       "text/plain": [
-       "True"
+       "tensor(2.9998, grad_fn=<MulBackward0>)"
       ]
      },
-     "execution_count": 20,
+     "execution_count": 32,
      "metadata": {},
      "output_type": "execute_result"
     }
    ],
    "source": [
-    "out1_train = quant_hard_tanh(inp1)\n",
-    "quant_hard_tanh.eval()\n",
-    "out2_eval = quant_hard_tanh(inp2)\n",
-    "assert out1_train.scale.isclose(out2_eval.scale).item()"
+    "out_channels = 3\n",
+    "inp3 = torch.rand(1, out_channels, 256, 256)  # (B, C, H, W)\n",
+    "inp3[:, 0, :, :] *= 3\n",
+    "\n",
+    "per_tensor_quant_relu = QuantReLU(return_quant_tensor=True)\n",
+    "out_tensor = per_tensor_quant_relu(inp3)\n",
+    "out_tensor.scale * ((2**8) -1)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can see that the per-tensor scale parameter has calibrated itself to provide a full quantization range of 3, matching that of the most extreme channel. \n",
```

Comment:
Most extreme -> highest? If I'm understanding the meaning correctly

Reply:
Yep, I'll make it more clear!

Reply:
I changed it to indicate the channel with largest dynamic range, instead of most "extreme".